server/svix-server/src/v1/endpoints/attempt.rs

```diff
@@ -174,47 +172,17 @@ pub struct ListAttemptsByEndpointQueryParameters {
     channel: Option<EventChannel>,
 }
 
-/// Fetches a list of [`MessageAttemptOut`]s for a given endpoint ID
-async fn list_attempts_by_endpoint(
-    Extension(ref db): Extension<DatabaseConnection>,
-    ValidatedQuery(mut pagination): ValidatedQuery<Pagination<MessageAttemptId>>,
-    ValidatedQuery(ListAttemptsByEndpointQueryParameters {
-        status,
-        status_code_class,
-        event_types,
-        channel,
-    }): ValidatedQuery<ListAttemptsByEndpointQueryParameters>,
-    Path((_app_id, endp_id)): Path<(ApplicationId, EndpointId)>,
-    AuthenticatedApplication {
-        permissions: _,
-        app,
-    }: AuthenticatedApplication,
-) -> Result<Json<ListResponse<MessageAttemptOut>>> {
-    let limit = pagination.limit;
-    let iterator = pagination.iterator.take();
-
-    // Confirm endpoint ID belongs to the given application
-    if endpoint::Entity::secure_find_by_id(app.id, endp_id.clone())
-        .one(db)
-        .await?
-        .is_none()
-    {
-        return Err(Error::Http(HttpError::not_found(None, None)));
-    }
-
-    let mut query = messageattempt::Entity::secure_find_by_endpoint(endp_id)
-        .order_by_desc(messageattempt::Column::Id)
-        .limit(limit + 1);
-
-    if let Some(iterator) = iterator {
-        query = query.filter(messageattempt::Column::Id.lt(iterator));
-    }
-
-    if let Some(status) = status {
+// Applies filters from the [`ListAttemptsByEndpointOrMsgQueryParameters`] of the
+// [`list_attempts_by_endpoint`] and [`list_attempts_by_msg`] endpoints that are common
+fn list_attempts_by_endpoint_or_message_query_filters(
+    mut query: Select<messageattempt::Entity>,
+    params: ListAttemptsByEndpointOrMsgQueryParameters,
```

Comment:
Pass the parameters explicitly, don't pass them as a structure like this. It's similar to what we talked about before, be explicit about what you pass, don't pass massive "context" objects.